vuestorefront/models/website_page.py

```diff
@@ -24,19 +26,18 @@ def prepare_vsf_domain(self, **kw):
     def render_vsf_page(self, **kw):
         self.ensure_one()
         website = get_website(self.env)
+        # NOTE. Even if page has no website set, we assume current
+        # website all the time for now.
+        base_url = website.get_base_url()
+        endpoint = urljoin(base_url, self.url)
         # TODO: make it possible to render page without its layout
         # directly, so we would not need to strip rendered content
         # afterwards.
-        content = self.env["ir.ui.view"]._render_template(
-            self.view_id.id,
-            {
-                'deletable': True,
-                'main_object': self,
-                'website': website,
-            }
-        ).decode()
-        content = self._postprocess_vsf_page_rendering(content, **kw)
-        return content
+        # We should also render template directly (e.g. using
+        # self.env["ir.ui.view"]._render_template) instead of doing
+        # external call.
```

Comment:
Maybe a comment what were the issues with that approach can clarify why we moved back from it to this.